src/components/sbb-footer/sbb-footer.tsx

```diff
@@ -7,67 +7,33 @@ import { InterfaceFooterAttributes } from './sbb-footer.custom';
   tag: 'sbb-footer',
 })
 export class SbbFooter {
-  /** Footer appearance */
-  @Prop() public appearance?: InterfaceFooterAttributes['appearance'] = 'primary';
+  /**
+   * Negative coloring variant flag
+   */
+  @Prop() public negative: boolean;
+
+  @Prop() public variant?: InterfaceFooterAttributes['variant'] = 'default';
 
   /** Footer title text, visually hidden,  necessary for screenreaders */
   @Prop() public accessibilityTitle!: string;
 
-  /** Host element */
-  @Element() private _hostElement: HTMLElement;
-
-  private _hasCol1Slot: boolean;
-  private _hasCol2Slot: boolean;
-  private _hasCol3Slot: boolean;
-  private _hasCol4Slot: boolean;
-  private _hasClockSlot: boolean;
-  private _hasBottomSlot: boolean;
-
-  public componentWillLoad(): void {
-    // Check slots
-    this._hasCol1Slot = Boolean(this._hostElement.querySelector('[slot="col-1"]'));
-    this._hasCol2Slot = Boolean(this._hostElement.querySelector('[slot="col-2"]'));
-    this._hasCol3Slot = Boolean(this._hostElement.querySelector('[slot="col-3"]'));
-    this._hasCol4Slot = Boolean(this._hostElement.querySelector('[slot="col-4"]'));
-    this._hasClockSlot = Boolean(this._hostElement.querySelector('[slot="clock"]'));
-    this._hasBottomSlot = Boolean(this._hostElement.querySelector('[slot="bottom"]'));
-  }
-
   public render(): JSX.Element {
-    const className = `footer footer--${this.appearance}`;
+    const variantClass = this.variant !== 'default' ? ' footer--clock-columns' : '';
```

Comment:
Instead the adding classes, (if possible) consider using reflect: true in the prop definition, then change the scss accordingly. Example: https://github.com/lyne-design-system/lyne-components/blob/master/src/components/sbb-card/sbb-card.scss